File: apple/Sources/ReactNativeWatchOS/Modifiers/AccessibilityIdentifier.swift
import SwiftUI

/// SwiftUI `.accessibilityIdentifier(_:)`. Sets a non-localized identifier
/// used by UI tests; not spoken by VoiceOver.
enum RNWAccessibilityIdentifierModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("accessibilityIdentifier") { view, params, _ in
            AnyView(view.accessibilityIdentifier(params.string("identifier") ?? ""))
        }
    }
}

File: src/modifiers/accessibilityIdentifier.ts
import { createModifier } from './createModifier';

/// SwiftUI `.accessibilityIdentifier(_:)`. Sets a non-localized identifier
/// used by UI tests to locate the view; not spoken by VoiceOver.
export function accessibilityIdentifier(identifier: string) {
  return createModifier('accessibilityIdentifier', { identifier });
}
